types/jscodeshift/src/Collection.d.ts

```diff
@@ -124,7 +126,7 @@ export function fromNodes(...args: any[]): any;
  * @param methods Methods to add to the prototype
  * @param type Optional type to add the methods to
  */
-export function registerMethods(methods: object, type?: types.Type<any>): void;
+export function registerMethods<N extends Core.ASTNode>(methods: Record<string, Method<N>>, type?: types.Type<N>): void;
```

Comment:
How would this type parameter be provided? There doesn't seem to be any way this is going to be inferred to anything besides `Core.ASTNode`.